### DISCUSSION TITLE
Any interest is supporting TLS keyfiles with passwords?

### DISCUSSION THREAD

**Discussion by idahogray** [celery/celery#7164] [Ideas]
# Checklist

* [x] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical enhancement to an existing feature.
* [x] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed enhancements.
* [x] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the if the same enhancement was already implemented in the
  master branch.
* [x] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues
* None

#### Possible Duplicates
* None

# Brief Summary

Rabbitmq supports verifying client's using a TLS keyfile. Celery supports this with one exception, the keyfile cannot be password
protected. I believe the underlying Python SSL library does support this so I am wondering if there is interest in adding that support
in Celery?

# Design
## Architectural Considerations

I believe Kombu would need updates as well as Celery.

## Proposed Behavior

I propose adding a new element to the `broker_use_ssl` dict to provide the keyfile's password.
`{"keyfile_password": "some strong password"}`.

## Proposed UI/UX

I believe only the `broker_use_ssl` setting would need updated from a UI standpoint.

## Diagrams

N/A

## Alternatives

The current feature works well, it just doesn't allow the keyfile to have a password. Therefore, a viable alternative is to
leave it the way it is.

[a comment by an automated account is not shown]

**Comment by bkovacev** [comment]
@auvipy Hello, I'd like to work on this feature for celery as I'd love to incorporate it in my project. We'd probably need to update:
https://github.com/celery/py-amqp/
https://github.com/celery/kombu/
and then the actual celery library.

Would I have the green light for doing this?

**Reply by auvipy** [reply]
You can come with POC
